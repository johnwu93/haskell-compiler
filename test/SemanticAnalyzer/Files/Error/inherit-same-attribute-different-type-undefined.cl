class Bar inherits Foo {
  x : X;
  bar(): Int {
     "String" + 1
  };
};

class Foo {
  x : String;
};

class Baz inherits Foo {
  x : X;
  bar(): Int {
    not 1
  };
};

class Main inherits Bar {
   main (): Int {
     x
   };
};
