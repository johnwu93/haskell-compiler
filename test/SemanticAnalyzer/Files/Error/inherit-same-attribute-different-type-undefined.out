inheritSameAttributeDifferentTypeUndefined.cl:1: Attribute x is an attribute of an inherited class.
inheritSameAttributeDifferentTypeUndefined.cl:1: Attribute x is an attribute of an inherited class.
inheritSameAttributeDifferentTypeUndefined.cl:1: Class X of attribute x is undefined.
inheritSameAttributeDifferentTypeUndefined.cl:1: non-Int arguments: String + Int
inheritSameAttributeDifferentTypeUndefined.cl:1: Inferred return type String of method main does not conform to declared return type Int.
inheritSameAttributeDifferentTypeUndefined.cl:1: Class X of attribute x is undefined.
inheritSameAttributeDifferentTypeUndefined.cl:1: Argument of 'not' has type Int instead of Bool.
inheritSameAttributeDifferentTypeUndefined.cl:1: Inferred return type Bool of method bar does not conform to declared return type Int.
Compilation halted due to static semantic errors.
